src/main/java/de/tum/in/www1/artemis/web/rest/tutorialgroups/TutorialGroupFreePeriodResource.java

```diff
@@ -109,8 +109,10 @@ public ResponseEntity<TutorialGroupFreePeriod> update(@PathVariable Long courseI
         updatedFreePeriod.setId(existingFreePeriod.getId());
         updatedFreePeriod.setTutorialGroupsConfiguration(configuration);
         updatedFreePeriod.setReason(tutorialGroupFreePeriod.reason);
-        updatedFreePeriod.setStart(interpretInTimeZone(tutorialGroupFreePeriod.date, START_OF_DAY, configuration.getCourse().getTimeZone()));
-        updatedFreePeriod.setEnd(interpretInTimeZone(tutorialGroupFreePeriod.date, END_OF_DAY, configuration.getCourse().getTimeZone()));
+        updatedFreePeriod.setStart(
+                interpretInTimeZone(tutorialGroupFreePeriod.startDate.toLocalDate(), tutorialGroupFreePeriod.startDate.toLocalTime(), configuration.getCourse().getTimeZone()));
```

Comment:
What happens if the end dateTime is before the start dateTime?

Reply:
I added a server-sided Check to prevent this from happening. Thanks for the heads up!